staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -376,6 +379,9 @@ func (wc *watchChan) prepareObjs(e *event) (curObj runtime.Object, oldObj runtim
 func decodeObj(codec runtime.Codec, versioner storage.Versioner, data []byte, rev int64) (runtime.Object, error) {
 	obj, err := runtime.Decode(codec, []byte(data))
 	if err != nil {
+		if fatalOnDecodeError {
+			panic(err) // we are running in a test environment and thus an error here is due to a coder mistake
```

Comment:
We want to enable this in test mode so we don't silently fail on colocated resources encoding incorrectly (as we do today). This is like the cache mutation detector, which only runs in test mode, and panics if you mutate an object in a shared informer cache